cwrc_audit_report.rb

```diff
@@ -95,23 +95,21 @@ module CWRCPerserver
   # TODO: use CSV gem
   # CSV header
   puts "cwrc_pid (#{cwrc_objs.count}),"\
-    "cwrc_mtime (#{Time.now}),"\
+    "cwrc_mtime (#{Time.now.iso8601}),"\
     "swift_id (#{swift_container.container_metadata[:count]}),"\
     'swift_timestamp,swift_bytes,status'
 
   # TODO: find a better way to merge CWRC and Swift hashes into an output format
   # for each cwrc object
   cwrc_objs&.each do |cwrc_obj|
-    cwrc_pid = cwrc_obj['pid'].to_s
-    cwrc_mtime = cwrc_obj['timestamp'].to_s
-    swift_id = cwrc_pid.dup
+    cwrc_pid = cwrc_obj['pid']
+    cwrc_mtime = cwrc_obj['timestamp']
     # account for the CWRC PID ':' replaced with "_" in the Swift ID
-    swift_id.sub! ':', '_'
+    swift_id = cwrc_pid.sub! ':', '_'
```

Comment:
Just to double-check, `sub!` here will mutate `cwrc_pid` and return a reference to the same string to `swift_id` (they're essentially just two names for the same thing in this scope, which might lead to some errors down the road).

Is this what you were going for? If not, you could use `sub` instead of `sub!` which would *not* change `cwrc_pid` and instead create a new string for `swift_id`.

Otherwise I'd suggest just using one variable throughout.

Reply:
No mutate.